dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -603,6 +605,189 @@ def test_correlation_update(self):
         np.testing.assert_array_almost_equal(expected_corr_mat,
                                              profiler.correlation_matrix)
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler',
+                spec=StructuredDataLabeler)
+    def test_chi2(self, *mocks):
+        # Empty
+        data = pd.DataFrame([])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.chi2_matrix)
+
+        # Single column
+        data = pd.DataFrame({'a': ["y", "y", "n", "n", "y"]})
+        profiler = dp.StructuredProfiler(data)
+        expected_mat = np.array([1])
+        self.assertEqual(expected_mat, profiler.chi2_matrix)
+
+        data = pd.DataFrame({'a': ["y", "y", "y", "y", "n", "n", "n"],
+                             'b': ["y", "maybe", "y", "y", "n", "n", "maybe"],
+                             'c': ["n", "maybe", "n", "n", "n", "y", "y"]})
```

Comment:
do we deal with non values in this PR?

Reply:
non values? the categorical profiler should not receive nan values if that's what you mean.

Reply:
Agreed with Jeremy, NaN values shouldn't affect the chi2 calculation.